dashboard/app/controllers/api/v1/pd/applications_controller.rb

```diff
@@ -71,11 +71,22 @@ def quick_view
   # GET /api/v1/pd/applications/cohort_view?role=:role&regional_partner_filter=:regional_partner_name
   def cohort_view
     applications = get_applications_by_role(params[:role].to_sym).where(status: ['accepted', 'withdrawn'])
+    cohort_capacity = nil
 
     unless params[:regional_partner_filter].nil? || params[:regional_partner_filter] == 'all'
       applications = applications.where(regional_partner_id: params[:regional_partner_filter] == 'none' ? nil : params[:regional_partner_filter])
     end
 
+    unless ['none', 'all'].include? params[:regional_partner_filter]
+      partner_id = params[:regional_partner_filter] ? params[:regional_partner_filter] : current_user.regional_partners.first
+      partner = RegionalPartner.find_by(id: partner_id)
+      if params[:role] == 'csd_teachers'
+        cohort_capacity = partner.cohort_capacity_csd
+      elsif params[:role] == 'csp_teachers'
+        cohort_capacity = partner.cohort_capacity_csp
+      end
+    end
```

Comment:
nit: Let's put this is a method, e.g. `get_partner_cohort_capacity` so the action is less cluttered, and add some basic tests.

Then line 101 below can simply reference it like `capacity: get_partner_cohort_capacity`